src/main/kotlin/lotto/domain/collection/LottoTickets.kt

```diff
@@ -0,0 +1,22 @@
+package lotto.domain.collection
+
+import lotto.domain.LottoTicket
+
+data class LottoTickets(
+    private val lottoTickets: List<LottoTicket>
+) {
+    init {
+        require(lottoTickets.isNotEmpty()) {
+            "로또 티켓 개수가 1개 이상이여야 합니다."
+        }
+    }
+
+    fun getLottoTickets(): List<LottoTicket> {
+        return lottoTickets
+    }
+
+    fun getMatchCount(matchCount: Int, numbers: Set<Int>): Int {
+        return lottoTickets.filter { it.numbers.intersect(numbers).size == matchCount }
+            .size
```

Comment:
`LottoTicket`에게 적절한 형태의 역할을 위임하고, `count`를 써볼까요?
```suggestion
        return lottoTickets.count { it.match(numbers) == matchCount }
```

Reply:
반영하였습니다.
- [일급 컬랙션 객체 개선 커밋](https://github.com/next-step/kotlin-lotto/pull/350/commits/bc78d1f85599c5a543033c933ca8ecb1c426b146)
- [책임 분리 커밋](https://github.com/next-step/kotlin-lotto/pull/350/commits/c39dfa3025253073abd3c8abe8b63d2b1240a6fd)